__tests__/core/gatherer.test.ts

```diff
@@ -116,4 +117,25 @@ describe('Gatherer', () => {
       await Gatherer.stop();
     });
   });
+
+  describe('Network emulation', () => {
+    it('applies network throttling', async () => {
```

Comment:
would add one more test to check the network condition on subsequent window or popup.